src/graphnet/models/components/layers.py

```diff
@@ -593,3 +602,393 @@ def forward(
             )
             x = x + self.drop_path(self.gamma_2 * self.mlp(self.norm2(x)))
         return x
+
+
+class GritSparseMHA(LightningModule):
+    """Proposed Attention Computation for GRIT.
+
+    Original code:
+    https://github.com/LiamMa/GRIT/blob/main/grit/layer/grit_layer.py
+    """
+
+    def __init__(
+        self,
+        in_dim: int,
+        out_dim: int,
+        num_heads: int,
+        use_bias: bool,
+        clamp: float = 5.0,
+        dropout: float = 0.0,
+        activation: nn.Module = nn.ReLU,
+        edge_enhance: bool = True,
+    ):
+        """Construct 'GritSparseMHA'.
+
+        Args:
+            in_dim: Dimension of the input tensor.
+            out_dim: Dimension of theo output tensor.
+            num_heads: Number of attention heads.
+            use_bias: Apply bias the key and value linear layers.
+            clamp: Clamp the absolute value of the attention scores to a value.
+            dropout: Dropout layer probability.
+            activation: Activation function.
```

Comment:
Could mention here that the activation function is assumed to be un-instantiated. E.g.
```
"""
activation:  Reference to uninstantiated activation function. E.g. `torch.nn.LeakyReLU`
"""
```